community/dbms/src/main/java/org/neo4j/commandline/dbms/MemoryRecommendationsCommand.java

```diff
@@ -199,6 +219,92 @@ public void execute( String[] args ) throws IncorrectUsage
         print( initialHeapSize.name() + "=" + heap );
         print( maxHeapSize.name() + "=" + heap );
         print( pagecache_memory.name() + "=" + pagecache );
+
+        boolean db = arguments.getBoolean( ARG_DB );
+        if ( !db )
+        {
+            return;
+        }
+        File configFile = configDir.resolve( Config.DEFAULT_CONFIG_FILE_NAME ).toFile();
+        File storeDir = getConfig( configFile ).get( database_path );
+
+        print( "#" );
+        print( "# A minimal page cache memory setting to fully map the database contents of '" + storeDir.getAbsolutePath() + "':" );
+        print( pagecache_memory.name() + "=" + dbSpecificPageCacheMemory( storeDir ) );
+    }
+
+    private FilenameFilter getNativeIndexFileFilter()
+    {
+        return ( dir, name ) ->
+        {
+            File file = new File( dir, name );
+            if ( FailureStorage.DEFAULT_FAILURE_FILE_NAME.equals( name ) )
+            {
+                return false;
+            }
+
+            if ( outsideWorld.fileSystem().isDirectory( file ) && dir.getName().startsWith( "lucene-" ) )
+            {
+                // Do not go down directory containing lucene index files, i.e. something like lucene-x.y/z/index.file
+                // so this condition is for not going down the 'z' directory. There may be spatial/temporal native index files
+                // in lucene/x.y/ so still go down there.
+                return false;
+            }
+            return true;
+        };
+    }
+
+    private String dbSpecificPageCacheMemory( File storeDir )
+    {
+        long total = 0;
+
+        // All neostore files
+        for ( StoreType type : StoreType.values() )
+        {
+            if ( type.isRecordStore() )
+            {
+                File file = new File( storeDir, type.getStoreFile().storeFileName() );
+                if ( outsideWorld.fileSystem().fileExists( file ) )
+                {
+                    total += outsideWorld.fileSystem().getFileSize( file );
+                }
+            }
+        }
+
+        // All native index files
+        File baseSchemaIndexDir = baseSchemaIndexFolder( storeDir );
+        total = sumNativeIndexFiles( baseSchemaIndexDir, getNativeIndexFileFilter(), total );
```

Comment:
This could be `total += …` and then there's no need to pass `total` as an argument to the method.